### ISSUE TITLE
RFC: Deprecate  `auto_select_gpus` Trainer argument

### ISSUE THREAD

**Issue by awaelchli** [PyTorchLightning/pytorch-lightning#13079]
## Proposed refactor

Deprecate the `Trainer(auto_select_gpus=True|False)` option and just enable it always when `devices=int` gets selected. 

### Motivation

- The feature is misleading, we don't actually have an algorithm to select the gpus automatically in a smart way. A confusion by user was raised recently: #13012
- The flag only applies to the GPU accelerator. There are no equivalent flags for tpus, ipus, etc.
- What auto_select_gpus does is such a niche use case that it's almost not worth talking about. The implementation just runs through all available GPUs and tests if it can place a tensor in memory. This essentially tests wether the GPU is in exclusive mode or not. [See the implementation here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/auto_gpu_select.py).
- Internally, the flag is framed as "tuning" as the functions are under the tuner module, but this does not fall under the term "tuning" in my opinion.

### Pitch

Keep the feature, but simply enable it by default and remove the flag from the Trainer. It is still useful to have the check against exclusivity, for example on managed clusters, and I can't think of a reason why it would be undesired. This would only apply when passing `devices=int` and not when indices get passed. 

### Alternatives

Keep it and come up with sophisticated methods to determine whether a GPU should be selected or not based on memory profile, utilization etc. This IMO is not feasible because not enough information is available before training.

### Additional context

#13012

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

**Comment by tanmoyio** [comment]
@awaelchli want to work on it

**Comment by carmocca** [comment]
Sounds good to me.

Should we allow customizing it by passing `GPUAccelerator(auto_select=...)`?

**Comment by ananthsub** [comment]
FYI this also came up in https://github.com/PyTorchLightning/pytorch-lightning/issues/10535

**Comment by mogwai** [comment]
I agree that the name isn't generic and there has been a good trend in lightning the last few versions to address the naming not matching the accelerator. 

It makes sense that if  CUDA accelerator is being used devices have been specified that the Trainer would automatically find gpus that have 0% utilisation and 0% memory allocation from torch. This would then remove the need for auto_select_gpu param.

However, it still might be important for users to specify gpus they might want to use.
